### DISCUSSION TITLE
Development workflow

### DISCUSSION THREAD

**Discussion by madhan34223** [i3/i3#6345] [Q&A] [answered]
Hi,

I have built the i3 binary. How do you test it quickly without rebooting . 
What is the development workflow? Do you use xephyr or tigervnc ? 

Please share your development workflow.

Thanks

**Comment by orestisfl** [comment, marked as answer]
I have this script using xephyr: https://github.com/orestisfl/dotfiles/blob/master/xorg/start-xephyr.sh

**Comment by madhan34223** [comment]
Thanks. Since i3 running inside Xephyr does not grab the focus of super key. I assume you develop i3 in another window manager. Is that correct?

**Reply by orestisfl** [reply]
In Xephyr, ctrl+shift grabs focus + keyboard/mouse

**Reply by madhan34223** [reply]
That works well. Thanks.
